### PR TITLE
chore!: break out BlobTx and IndexWrapper types into separate tx package

### PR DESCRIPTION
**Pull request by cmwaters**
While updating `celestia-app`, it felt out of place that BlobTx and IndexWrapper should be part of the `shares` package. This moves them out

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,8 +7,10 @@
 Package   | Description
 ----------|---------------------------------------------------------------------------------------------------------------------
 inclusion | Package inclusion contains functions to generate the blob share commitment from a given blob.
+proto     | Package contains proto definitions and go generated code
```
Comment (rootulp) [inline README.md:10]: [question] does this proto package actually exist? It looks like the proto definition for BlobTx and IndexWrapper are still inside proto/blob/v1

**Review by rootulp — APPROVED** [review]
Do we also want to move the proto definitions for BlobTx and IndexWrapper from `proto/blob/v1/blob.proto` to something like `proto/tx/blobtx.proto` and `proto/tx/indexwrapper.proto`

File: README.md (hunk 1 above)
Comment (rootulp) [reply README.md:10]: put another way: there is no Go package named `proto`

**Review by rach-id — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (cmwaters) [reply README.md:10]: No but here is generated code in the proto directory and that's what I was referring to. I'm happy to remove it

**Comment by cmwaters** [comment]
> Do we also want to move the proto definitions for BlobTx and IndexWrapper from `proto/blob/v1/blob.proto` to something like `proto/tx/blobtx.proto` and `proto/tx/indexwrapper.proto`

Yeah it seems more logical to separate them but we don't need to

File: README.md (hunk 1 above)
Comment (rootulp) [reply README.md:10]: I thought this list was just Go packages but I can see an argument for expanding this list to all directories and not just Go packages

**Pull request merged**